toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -33,24 +33,32 @@ type BuildChannels struct {
 }
 
 // BuildRequest represents the results of a build agent trying to build a given node.
+// - Node: The node to process
+// - PkgGraph: The graph of all packages
+// - AncillaryNodes: All additional build nodes that share a common SRPM (will be processed together)
+// - CanUseCache: If the scheduler is allowed to use a cached/already built copy of this package instead of building it
+// - IsDelta: Is this a pre-downloaded rpm (not traditional cache) that we may be able to skip rebuilding
+// - ExpectedFreshness: The expected freshness of the node (used to determine if we can skip building future nodes)
 type BuildRequest struct {
-	Node           *pkggraph.PkgNode
-	PkgGraph       *pkggraph.PkgGraph
-	AncillaryNodes []*pkggraph.PkgNode
-	CanUseCache    bool
-	IsDelta        bool
+	Node              *pkggraph.PkgNode
+	PkgGraph          *pkggraph.PkgGraph
+	AncillaryNodes    []*pkggraph.PkgNode
+	CanUseCache       bool
+	IsDelta           bool
+	ExpectedFreshness int
 }
 
 // BuildResult represents the results of a build agent trying to build a given node.
 type BuildResult struct {
```

Comment:
Do we want to add comments similar to what you did for `BuildRequest`?

Reply:
Will do.